nemo/lightning/pytorch/strategies/megatron_strategy.py

```diff
@@ -135,7 +135,6 @@ class MegatronStrategy(DDPStrategy, io.IOMixin):
         ckpt_save_optimizer (bool): Save optimizer states in checkpoint. Defaults to True.
         ddp (Union[DDPLiteral, DistributedDataParallelConfig]): DDP configuration. Defaults to "megatron".
         lazy_init (bool): Use lazy initialization for model parallel parameters. Defaults to False.
-        pipeline_dtype (Optional[torch.dtype]): Data type for pipeline parallelism. Defaults to None.
```

Comment:
I think it would be good to have a bit better process for deprecating things. Could you add a deprecation notice, explaining that it will be removed at 25.01 (or whatever the first release nest year is) and that from now on it will be inferred from the precision-plugin?